internal/pkg/agent/application/paths/paths_unix.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !windows
+
+package paths
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"path/filepath"
+)
+
+func initialControlSocketPath(topPath string) string {
+	socketPath := filepath.Join(topPath, ControlSocketName)
+	unixSocket := fmt.Sprintf("unix://%s", socketPath)
+	if len(unixSocket) < 104 {
+		return unixSocket
+	}
+	// place in global /tmp to ensure that its small enough to fit; current path is way to long
+	// for it to be used, but needs to be unique per Agent (in the case that multiple are running)
+	return fmt.Sprintf(`unix:///tmp/elastic-agent/%x.sock`, sha256.Sum256([]byte(socketPath)))
```

Comment:
Can you use the code that already exists for this?

https://github.com/elastic/elastic-agent/blob/1410d2a171015456174663ac64b06d26b8ad4509/pkg/utils/socket_unix.go#L21-L44

Same goes for Windows.

Reply:
Updated.